### PR TITLE
NIM-11 Run test coverage and send the results to codecov.io

### PR DESCRIPTION
**Pull request by romainrichard**
We're now going to be able to see how our test coverage evolves over time thanks to [codecov.io](https://codecov.io/gh/hacksaurz/nim). Not much data there yet because I just enabled the integration, but soon it should look something like [this](https://codecov.io/gh/tox-dev/tox-travis).

Not running `codecov` in `tox` because you need a token to upload the results, while you don't need one when it's running on Travis. Note though that running `tox` will show the test coverage, so you have the data, it just won't upload it.

Had to rename the environments in `tox` because `tox-travis` looks for environments based on the python versions we give it. In this case, in `.travis.yml`, we specified `"3.6"` so it'll look for environments that contain `py36`. It makes `tox.ini` a bit uglier but on the bright side, now Travis actually runs our tests!

We now have 2 badges in our README :tada:

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
 commands = flake8 --max-line-length 99 --max-complexity 12 nim
 deps = flake8
 
-[testenv:test]
+[testenv:py36-test]
+basepython = python3.6
 use_develop = True
 commands =
-    pytest {posargs}
+    pytest --cov=./ {posargs}
```
Comment (dodecaphonix) [inline tox.ini]: How does `posargs` work when running `tox` locally?

Comment (romainrichard) [reply tox.ini]: You wrote this, you tell me.

Comment (dodecaphonix) [reply tox.ini]: Hahaha, my bad. I thought it had to do with `--cov`.

File: tox.ini [hunk 2]
```diff
@@ -1,26 +1,26 @@
 [tox]
-envlist = lint, test
+envlist = py36-lint, py36-test
```
Comment (dodecaphonix) [inline tox.ini]: There's no way to get Travis to play nice with the original names?

Comment (romainrichard) [reply tox.ini]: Maybe there is. Feel free to try, I'm just happy I got it to work.
Plus having this format probably prepares us for supporting other Python versions, so I doubt we'll be able to keep it the way it was.

File: .travis.yml [hunk 3]
```diff
@@ -2,8 +2,10 @@ sudo: false
 language: python
 python:
 - '3.6'
-install: pip install tox-travis
+install: pip install codecov tox-travis
 script: tox
+after_success:
+  - codecov
 notifications:
   slack:
```
Comment (daveLambe) [inline .travis.yml]: How come `3.6` & `codecov` have dashes but `slack` doesn't?

Comment (romainrichard) [reply .travis.yml]: list versus dict?

**Pull request merged**